webrender/src/gpu_cache.rs

```diff
@@ -844,6 +894,57 @@ impl GpuCache {
         }
     }
 
+    #[cfg(not(feature="gleam"))]
+    pub fn write_updates<B: hal::Backend>(
+        &mut self,
+        buffer: &mut PersistentlyMappedBuffer<B>,
+        device: &B::Device,
+        old_buffer: Option<PersistentlyMappedBuffer<B>>,
+        send_buffer: bool,
+    ) -> GpuCacheBufferUpdate<B> {
+        let mut address_max = 0;
+        {
+            let (mut mapped_range, size) = buffer.map(device, None);
+            let mut writer = unsafe {
+                    mapped_range.write::<GpuBlockData>(
+                    device,
+                    0..(size / mem::size_of::<GpuBlockData>() as u64)
+                )
+            }.unwrap();
+            let writer_slice = unsafe { writer.slice() };
+
+            let blocks = mem::replace(&mut self.texture.pending_blocks, Vec::new());
+            for update in mem::replace(&mut self.texture.updates, Vec::new()) {
+                match update {
+                    GpuCacheUpdate::Copy {
+                        block_index,
+                        block_count,
+                        address,
+                    } => {
+                        let address = address.v as usize * MAX_VERTEX_TEXTURE_WIDTH + address.u as usize;
+                        address_max = address_max.max((address + block_count) as u64 * GpuBlockData::SIZE);
+                        for i in 0 .. block_count {
+                            writer_slice[address + i] = blocks[block_index + i];
```

Comment:
can we use `copy_from_slice`?

Reply:
Yes we can, changed it here https://github.com/szeged/webrender/pull/307/files#diff-950eebe8d7a83d7c04825debe3b9b68fR927 and here https://github.com/szeged/webrender/pull/307/files#diff-f5062b694b9fe53fc1757ed483d577d9R928